### ISSUE TITLE
Partial training when transfer learning?

### ISSUE THREAD

**Issue by mlanas** [instadeepai/Mava#316]
Hi, I'm doing some transfer(curriculum) learning with a custom model. The custom Sonnet module is made of 4 MLPs. What I would like to do is only train one of those. So after loading the weights from the checkpoint of the first training, I want the trainer to only update the weights of one of the MLPs, while the other 3 will use the loaded weights and never be updated. **Is this possible in Mava?**

**Comment by DriesSmit** [comment]
Hey @mlanas. I think it would be possible. This seems to be more of a Tensorflow technicality. What you can do is wrap the trainer's [_backward](https://github.com/instadeepai/Mava/blob/develop/mava/systems/tf/maddpg/training.py#L474) function. This function is where the gradients are calculated. You will then have to go through the `policy_variables` and filter out the variables of the layers you don't want to update. This [issue](https://stackoverflow.com/questions/37326002/is-it-possible-to-make-a-trainable-variable-not-trainable) proposes something like: 
```
tvars = tf.trainable_variables()
g_vars = [var for var in tvars if 'g_' in var.name] 
```

Hope that helps.
